src/core/providers/ogr/qgsogrprovider.cpp

```diff
@@ -1920,13 +1927,33 @@ bool QgsOgrProvider::addFeatures( QgsFeatureList &flist, Flags flags )
 
   const bool inTransaction = startTransaction();
 
+  QgsFeatureId incrementalFeatureId = -1;
+  OGRGeometryH filter;
+  if ( !( flags & QgsFeatureSink::FastInsert ) &&
+       ( mGDALDriverName == QLatin1String( "CSV" ) || mGDALDriverName == QLatin1String( "XLSX" ) || mGDALDriverName == QLatin1String( "ODS" ) ) )
+  {
+    filter = mOgrLayer->GetSpatialFilter();
```

Comment:
this should be the attribute filter and not the spatial filter, right?

Reply:
@nyalldawson , correct.

Reply:
@nyalldawson , resolved in the follow up commit I've just added to this backport PR.